lib/user_stats.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require_relative 'bus'
+
+module UserStats
+  K = 40.0
+  K_NEW = 60.0
+  YEARS_INCLUDED = 4
+  BATCH_SIZE = 1000
+
+  def self.calculate_stats
+    user_stats = Hash.new { |hash, key| hash[key] = Hash.new([1200, 0]) }
+    on_or_after = Date.today - (365 * YEARS_INCLUDED)
+    offset = 0
+
+    loop do
```

Comment:
i think there's a sequel method that let's you iterate in batches. checkout paged_each(rows_per_fetch: 1000)

Reply:
Awesome. Fixed.